SPEC.md

```diff
@@ -143,17 +143,32 @@ When a build tool reads an artifact, it should for each input file:
 Concatenate all records together and apply lexical ordering by line to create the GitBOM document of the
 output artifact.
 
-### GitBOM Document persistence by a Build Tool
+## Annex A
 
-When persisting an output artifact to a file system, the build tool should create a subdirectory of the directory to which the output artifact is being written of the form:
-```.bom/objects/``` and write out the GitBOM document to ```.bom/objects/${gitoid:0:2}/${gitoid:2:}``` where ```${gitoid}```
-is the git object id of the GitBOM document.
+Annex A documents known methods of persisting GitBOM Documents to various stores.
+### GitBOM Document persistence by a Build Tool to its local filesystem
 
-### Build tool persistence of related metadata
+If a build tool persists GitBOM information to its local filesystem, the build tool should write out the GitBOM Document to ```${GITBOM_DIR}/objects/${Artifact Identifier Type uri prefix with ':' replaced by '/'}/${GitBOMID:0:2}/${GitBOMID:2:}``` where ```${artifact id}``` is the GitBOM id in lowercase hexidecimal.
 
-When persisting an output artifact to a file system, if the build tool has additional metadata to persist that makes reference to the Artifact Dependency Graph (ADG),
-it should persist that metadata to a subdirectory of the directory to which the output artifact is being written of the form: ```.bom/metadata/${tool}/```.  Filenaming and subdirectory structure below that point is at the discretion of the build tool.
-## Annex A
+Example:
+
+```
+.adg/objects/gitoid/blob/sha1/0e/8efd4cdf0d5bafcfcae658c2662a73b199b301
+```
+
+#### Build tool persistence of related metadata
+
+If the build tool has additional metadata to persist that makes referece to the Artifact Dependency Graph (ADG),
+it should persist that metadata to a subdirectory of the directory to which the output artifact is being written of the form: ```${GITBOM_DIR}/metadata/${tool}/```.  Filenaming and subdirectory structure below that point is at the discretion of the build tool.
+
+#### Build tool selection of GITBOM_DIR
+GITBOM_DIR should be set in order of precedence by:
+1.  A non-empty env variable named GITBOM_DIR
+2.  A build tool specific flag
+3.  A subdirectory ```.adg/``` of the directory to which it is writing out the artifact
```

Comment:
Do we want to care about XDG directory spec issues for this? I know there are folks who are pretty sensitive to where things get placed on their filesystem and particularly in their home directory, and would definitely want GitBOM to respect their XDG directory configuration, if set.

Reply:
This of course gets into platform-specific issues (Windows doesn't follow the XDG spec, for example).

For prior art on "where do I put these files," I like the Rust `dirs` crate, which documents where the three major platforms put different kinds of files: https://docs.rs/dirs/4.0.0/dirs/

Reply:
Realizing I misunderstood the sentence to which I am commenting, which is as a subdirectory of the artifact location, and not a `HOME`-relative directory. So perhaps this is moot!